### PR TITLE
Backport 'release/2024.10-rc1'

### PR DESCRIPTION
**Pull request by fedemkr**
## 📔 Objective

Merge `release/2024.10-rc1` into `main` branch with merge conflicts fixed.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by matt-livefront — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -23,7 +23,6 @@ on:
       compiler-flags:
         description: "Compiler Flags - e.g. 'DEBUG_MENU FEATURE2'"
         type: string
-        default: "DEBUG_MENU"
```
Comment (phil-livefront) [inline .github/workflows/build.yml:26]: 🤔 just want to make sure this change is intended?

Comment (fedemkr) [reply .github/workflows/build.yml:26]: I would think so @vvolkgang could you confirm?

Comment (vvolkgang) [reply .github/workflows/build.yml:26]: We need it in main but for visibility, let's revert it from this merge and I'll open a PR with this change targeting main.

Comment (fedemkr) [reply .github/workflows/build.yml]: Btw nice catch Phil and it has been reverted Álison 😄

**Review by phil-livefront — APPROVED** [review]

**Pull request merged**